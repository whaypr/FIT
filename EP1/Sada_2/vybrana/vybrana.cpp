#include <iostream>
#include <vector>
#include<bits/stdc++.h>

using namespace std;

int main() {
	int iterations;
	cin >> iterations;

	for (int i = 0; i < iterations; ++i) {
		int next_number;
		vector<int> selected; // index: seq len, value: last item of seq of given len

		while (true) {
			cin >> next_number;
			if (next_number == 0) break;

			// there is new longer selected seq -> push new item into vector
			if (selected.empty() || next_number >= selected.back())
				selected.push_back(next_number);
			else {
				auto it = upper_bound(selected.begin(), selected.end(), next_number); // find first num > next_number
				if (it == selected.end())
					selected.back() = next_number; // update last number
				else 
					*it = next_number; // update any other number
			}
		}

		cout << selected.size() << endl;
	}

	return 0;
}
