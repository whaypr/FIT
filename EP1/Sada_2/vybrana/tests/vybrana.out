8000
1
8000
172
161
41
141
89
151
101
170
107
52
122
89
147
73
151
145
92
169
172
22
114
38
113
111
122
176
169
98
119
66
26
96
126
66
100000
